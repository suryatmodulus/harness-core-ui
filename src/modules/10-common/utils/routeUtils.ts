/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type {
  AccountPathProps,
  OrgPathProps,
  ProjectPathProps,
  PipelinePathProps,
  TriggerPathProps,
  ExecutionPathProps,
  ConnectorPathProps,
  SecretsPathProps,
  PipelineType,
  FeatureFlagPathProps,
  SegmentPathProps,
  CVDataSourceTypePathProps,
  BuildPathProps,
  EnvironmentPathProps,
  DelegatePathProps,
  DelegateConfigProps,
  InputSetPathProps,
  VerificationPathProps,
  TargetPathProps,
  ModulePathParams,
  RolePathProps,
  ResourceGroupPathProps,
  UserGroupPathProps,
  UserPathProps,
  ServiceAccountPathProps,
  ServicePathProps,
  TemplateStudioPathProps
} from '@common/interfaces/RouteInterfaces'

export const accountPathProps: AccountPathProps = {
  accountId: ':accountId'
}

export const orgPathProps: OrgPathProps = {
  ...accountPathProps,
  orgIdentifier: ':orgIdentifier'
}

export const projectPathProps: ProjectPathProps = {
  ...orgPathProps,
  projectIdentifier: ':projectIdentifier'
}

export const pipelinePathProps: PipelinePathProps = {
  ...projectPathProps,
  pipelineIdentifier: ':pipelineIdentifier'
}

export const templatePathProps: TemplateStudioPathProps = {
  ...projectPathProps,
  templateIdentifier: ':templateIdentifier',
  templateType: ':templateType'
}

export const inputSetFormPathProps: InputSetPathProps = {
  ...pipelinePathProps,
  inputSetIdentifier: ':inputSetIdentifier'
}

export const triggerPathProps: TriggerPathProps = {
  ...pipelinePathProps,
  triggerIdentifier: ':triggerIdentifier'
}

export const executionPathProps: ExecutionPathProps = {
  ...pipelinePathProps,
  executionIdentifier: ':executionIdentifier'
}

export const connectorPathProps: ConnectorPathProps = {
  connectorId: ':connectorId'
}
export const verificationPathProps: VerificationPathProps = {
  verificationId: ':verificationId'
}

export const secretPathProps: SecretsPathProps = {
  secretId: ':secretId'
}

export const rolePathProps: RolePathProps = {
  roleIdentifier: ':roleIdentifier'
}

export const userGroupPathProps: UserGroupPathProps = {
  userGroupIdentifier: ':userGroupIdentifier'
}

export const userPathProps: UserPathProps = {
  userIdentifier: ':userIdentifier'
}

export const serviceAccountProps: ServiceAccountPathProps = {
  serviceAccountIdentifier: ':serviceAccountIdentifier'
}

export const resourceGroupPathProps: ResourceGroupPathProps = {
  resourceGroupIdentifier: ':resourceGroupIdentifier'
}

export const delegatePathProps: DelegatePathProps = {
  delegateIdentifier: ':delegateIdentifier'
}

export const delegateConfigProps: DelegateConfigProps = {
  delegateConfigIdentifier: ':delegateConfigIdentifier'
}

export const modulePathProps: ModulePathParams = {
  module: ':module'
}

export const pipelineModuleParams: Record<keyof PipelineType<unknown>, 'ci' | 'cd' | 'cf' | ':module'> = {
  module: ':module'
}

export const featureFlagPathProps: FeatureFlagPathProps = {
  featureFlagIdentifier: ':featureFlagIdentifier'
}

export const cvDataSourceTypePathProps: CVDataSourceTypePathProps = {
  dataSourceType: ':dataSourceType'
}

export const buildPathProps: BuildPathProps = {
  ...projectPathProps,
  buildIdentifier: ':buildIdentifier'
}

export const environmentPathProps: EnvironmentPathProps = {
  environmentIdentifier: ':environmentIdentifier'
}

export const segmentPathProps: SegmentPathProps = {
  segmentIdentifier: ':segmentIdentifier'
}

export const targetPathProps: TargetPathProps = {
  targetIdentifier: ':targetIdentifier'
}

export const servicePathProps: ServicePathProps = {
  serviceId: ':serviceId'
}

export function withAccountId<T>(fn: (args: T) => string) {
  return (params: T & { accountId: string }): string => {
    const path = fn(params)

    return `/account/${params.accountId}/${path.replace(/^\//, '')}`
  }
}

export function withOrgIdentifier<T>(fn: (args: T) => string) {
  return (params: T & { orgIdentifier: string }): string => {
    const path = fn(params)

    return `/orgs/${params.orgIdentifier}/${path.replace(/^\//, '')}`
  }
}

export function withProjectIdentifier<T>(fn: (args: T) => string) {
  return (params: T & { projectIdentifier: string }): string => {
    const path = fn(params)

    return `/projects/${params.projectIdentifier}/${path.replace(/^\//, '')}`
  }
}

export const getScopeBasedRoute = ({
  scope: { orgIdentifier, projectIdentifier, module },
  path
}: {
  scope: Partial<ProjectPathProps & ModulePathParams>
  path: string
}): string => {
  if (module && orgIdentifier && projectIdentifier) {
    return `/${module}/orgs/${orgIdentifier}/projects/${projectIdentifier}/setup/${path}`
  } else if (orgIdentifier && projectIdentifier) {
    return `/home/orgs/${orgIdentifier}/projects/${projectIdentifier}/setup/${path}`
  } else if (orgIdentifier) {
    return `/settings/organizations/${orgIdentifier}/setup/${path}`
  }
  return `/settings/${path}`
}

export const returnUrlParams = (url: string): string => `?returnUrl=${url}`

export const validateReturnUrl = (url: string): boolean => {
  const decodedUrl = decodeURIComponent(url)

  if (decodedUrl.startsWith('/')) {
    return true
  }

  try {
    const validUrl = new URL(decodedUrl)
    return window.location.hostname === validUrl.hostname
  } catch (_e) {
    return false
  }
}

export const returnLaunchUrl = (url: string): string => `${window.location.pathname.replace(/\/ng\//, '/')}${url}`
