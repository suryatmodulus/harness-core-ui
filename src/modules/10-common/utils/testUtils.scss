/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
