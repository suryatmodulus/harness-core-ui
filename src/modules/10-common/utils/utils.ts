/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
