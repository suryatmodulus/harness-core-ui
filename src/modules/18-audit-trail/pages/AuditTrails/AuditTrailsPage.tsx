/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { DateRangePickerButton, Layout } from '@wings-software/uicore'
import { useParams } from 'react-router-dom'
import { Page } from '@common/exports'
import { useGetAuditEventList } from 'services/audit'
import { useStrings } from 'framework/strings'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import type { OrgPathProps } from '@common/interfaces/RouteInterfaces'
import type { AuditFilterProperties } from 'services/audit'
import { useMutateAsGet } from '@common/hooks'
import AuditTrailsFilters from '@audit-trail/components/AuditTrailsFilters'
import ScopedTitle from '@common/components/Title/ScopedTitle'
import { Scope } from '@common/interfaces/SecretsInterface'
import AuditTrailsListView from './views/AuditTrailsListView'
import AuditTrailsEmptyState from './audit_trails_empty_state.png'
import css from './AuditTrailsPage.module.scss'

const AuditTrailsPage: React.FC = () => {
  const { accountId, orgIdentifier } = useParams<OrgPathProps>()
  const [selectedFilterProperties, setSelectedFilterProperties] = useState<AuditFilterProperties>()
  const [page, setPage] = useState(0)
  const { getString } = useStrings()
  const [startDate, setStartDate] = useState<Date>(() => {
    const start = new Date()
    start.setDate(start.getDate() - 7)
    start.setHours(0, 0, 0, 0)
    return start
  })

  const [endDate, setEndDate] = useState<Date>(() => {
    const end = new Date()
    end.setHours(23, 59, 59, 999)
    return end
  })

  const onDateChange = (selectedDates: [Date, Date]): void => {
    setPage(0)
    setStartDate(selectedDates[0])
    setEndDate(selectedDates[1])
  }

  const {
    data: auditData,
    loading,
    error,
    refetch
  } = useMutateAsGet(useGetAuditEventList, {
    queryParams: {
      accountIdentifier: accountId,
      pageSize: 10,
      pageIndex: page
    },
    body: {
      scopes: [{ accountIdentifier: accountId, orgIdentifier }],
      ...selectedFilterProperties,
      filterType: 'Audit',
      startTime: startDate.getTime(),
      endTime: endDate.getTime()
    }
  })

  const auditTrailTitle = getString('common.auditTrail')
  return (
    <>
      <Page.Header
        title={
          <ScopedTitle
            title={{
              [Scope.ACCOUNT]: auditTrailTitle,
              [Scope.PROJECT]: auditTrailTitle,
              [Scope.ORG]: auditTrailTitle
            }}
          />
        }
        breadcrumbs={<NGBreadcrumbs />}
        // content={
        //   <Text margin={{ right: 'tiny' }} font={{ variation: FontVariation.SMALL }} color={Color.GREY_600}>
        //     {getString('auditTrail.externalDataText')}
        //     <a className={css.link} target="_blank" href="https://harness.io/docs/api/tag/Audits" rel="noreferrer">
        //       {` ${getString('auditTrail.auditLogAPI')}`}
        //       <Icon
        //         className={css.launchIcon}
        //         margin={{ left: 'tiny', bottom: 'tiny' }}
        //         color={Color.PRIMARY_7}
        //         name="launch"
        //         size={12}
        //       />
        //     </a>
        //   </Text>
        // }
      />
      <Page.SubHeader className={css.subHeaderContainer}>
        <Layout.Horizontal flex className={css.subHeader}>
          <DateRangePickerButton
            className={css.dateRange}
            initialButtonText={getString('common.last7days')}
            dateRangePickerProps={{ defaultValue: [startDate, endDate] }}
            onChange={onDateChange}
            renderButtonText={selectedDates =>
              `${selectedDates[0].toLocaleDateString()} - ${selectedDates[1].toLocaleDateString()}`
            }
          />
          <Layout.Horizontal flex>
            <AuditTrailsFilters
              applyFilters={(properties: AuditFilterProperties) => {
                setPage(0)
                setSelectedFilterProperties(properties)
              }}
            />
          </Layout.Horizontal>
        </Layout.Horizontal>
      </Page.SubHeader>
      <Page.Body
        className={css.pageBody}
        noData={{
          when: () => !auditData?.data?.content?.length,
          image: AuditTrailsEmptyState,
          imageClassName: css.emptyStateImage,
          messageTitle: getString('auditTrail.emptyStateMessageTitle'),
          message: getString('auditTrail.emptyStateMessage')
        }}
        error={(error as any)?.data?.message || error?.message}
        retryOnError={() => refetch()}
        loading={loading}
      >
        <AuditTrailsListView setPage={setPage} data={auditData?.data || {}} />
      </Page.Body>
    </>
  )
}

export default AuditTrailsPage
