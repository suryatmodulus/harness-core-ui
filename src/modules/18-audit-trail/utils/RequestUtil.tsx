/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@wings-software/uicore'
import uniqBy from 'lodash/uniqBy'
import type { AuditTrailFormType, ProjectSelectOption } from '@audit-trail/components/FilterDrawer/FilterDrawer'
import type { AuditEventDTO, AuditFilterProperties, ResourceDTO, ResourceScopeDTO } from 'services/audit'
import type { StringKeys } from 'framework/strings'

export const actionToLabelMap: Record<AuditEventDTO['action'], StringKeys> = {
  CREATE: 'created',
  UPDATE: 'auditTrail.actions.updated',
  RESTORE: 'auditTrail.actions.restored',
  DELETE: 'deleted',
  UPSERT: 'auditTrail.actions.updated',
  INVITE: 'auditTrail.actions.invited',
  RESEND_INVITE: 'auditTrail.actions.invite_resent',
  REVOKE_INVITE: 'auditTrail.actions.invite_revoked',
  ADD_COLLABORATOR: 'auditTrail.actions.added_collaborator',
  REMOVE_COLLABORATOR: 'auditTrail.actions.removed_collaborator',
  ADD_MEMBERSHIP: 'auditTrail.actions.added_membership',
  REMOVE_MEMBERSHIP: 'auditTrail.actions.removed_membership',
  REVOKE_TOKEN: 'auditTrail.actions.revoke_token'
}

export const moduleToLabelMap: Record<AuditEventDTO['module'], StringKeys> = {
  CD: 'common.module.cd',
  CE: 'common.module.ce',
  CF: 'common.module.cf',
  CV: 'common.module.cv',
  CI: 'common.module.ci',
  CORE: 'common.module.core',
  PMS: 'common.module.pms',
  TEMPLATESERVICE: 'common.module.templateService'
}

export const resourceTypeToLabelMapping: Record<ResourceDTO['type'], StringKeys> = {
  ORGANIZATION: 'orgLabel',
  PROJECT: 'projectLabel',
  USER_GROUP: 'common.userGroup',
  SECRET: 'secretType',
  RESOURCE_GROUP: 'common.resourceGroupLabel',
  USER: 'common.userLabel',
  ROLE: 'common.role',
  ROLE_ASSIGNMENT: 'common.roleAssignmentLabel',
  PIPELINE: 'common.pipeline',
  TRIGGER: 'common.triggerLabel',
  TEMPLATE: 'common.template.label',
  INPUT_SET: 'inputSets.inputSetLabel',
  DELEGATE_CONFIGURATION: 'delegate.delegateConfiguration',
  SERVICE: 'service',
  ENVIRONMENT: 'environment',
  DELEGATE: 'delegate.DelegateName',
  SERVICE_ACCOUNT: 'serviceAccount',
  CONNECTOR: 'connector',
  API_KEY: 'common.apikey',
  TOKEN: 'token',
  DELEGATE_TOKEN: 'common.delegateTokenLabel'
}

export const getFilterPropertiesFromForm = (formData: AuditTrailFormType, accountId: string): AuditFilterProperties => {
  const filterProperties: AuditFilterProperties = { filterType: 'Audit' }
  const { actions, modules, users, resourceType, organizations, projects } = formData
  if (actions) {
    filterProperties['actions'] = actions.map(action => action.value) as AuditFilterProperties['actions']
  }
  if (modules) {
    filterProperties['modules'] = modules.map(
      (module: MultiSelectOption) => module.value
    ) as AuditFilterProperties['modules']
  }

  if (users) {
    filterProperties['principals'] = users.map(user => ({
      type: 'USER',
      identifier: user.value
    })) as AuditFilterProperties['principals']
  }

  if (resourceType) {
    filterProperties['resources'] = resourceType.map(type => ({
      type: type.value
    })) as AuditFilterProperties['resources']
  }

  if (projects && projects.length > 0) {
    filterProperties['scopes'] = projects.map(projectData => ({
      projectIdentifier: projectData.value as string,
      accountIdentifier: accountId,
      orgIdentifier: projectData.orgIdentifier
    }))
  }

  if (organizations) {
    if (!filterProperties['scopes']) {
      filterProperties['scopes'] = organizations.map(org => ({
        accountIdentifier: accountId,
        orgIdentifier: org.value as string
      }))
    } else {
      organizations.forEach(org => {
        if (filterProperties['scopes']?.findIndex(scope => scope.orgIdentifier === org.value) === -1) {
          filterProperties['scopes'].push({
            accountIdentifier: accountId,
            orgIdentifier: org.value as string
          })
        }
      })
    }
  }

  return filterProperties
}

const getOrgAndProjects = (scopes: ResourceScopeDTO[]) => {
  const organizations: MultiSelectOption[] = []
  const projects: ProjectSelectOption[] = []
  scopes.forEach(scope => {
    if (scope.orgIdentifier) {
      if (scope.projectIdentifier) {
        projects.push({
          label: scope.projectIdentifier,
          value: scope.projectIdentifier,
          orgIdentifier: scope.orgIdentifier
        })
      }
      organizations.push({ label: scope.orgIdentifier, value: scope.orgIdentifier })
    }
  })
  return {
    organizations: uniqBy(organizations, org => org.value),
    projects
  }
}

export const getFormValuesFromFilterProperties = (
  filterProperties: AuditFilterProperties,
  getString: (key: StringKeys, vars?: Record<string, any>) => string
): AuditTrailFormType => {
  const formData: AuditTrailFormType = {}
  const { actions, modules, principals, scopes, resources } = filterProperties
  if (actions) {
    formData['actions'] = actions?.map(action => ({ label: getString(actionToLabelMap[action]), value: action }))
  }

  if (modules) {
    formData['modules'] = modules?.map(module => ({ label: getString(moduleToLabelMap[module]), value: module }))
  }

  if (principals) {
    formData['users'] = principals?.map(principal => ({
      label: principal.identifier,
      value: principal.identifier
    }))
  }

  if (resources) {
    formData['resourceType'] = resources?.map(resource => ({
      label: getString(resourceTypeToLabelMapping[resource.type]),
      value: resource.type
    }))
  }

  return {
    ...formData,
    ...(scopes ? getOrgAndProjects(scopes) : {})
  }
}

export const formToLabelMap = (obj: Record<string, any>) => {
  const labelMap: {
    [key: string]: any
  } = {}
  Object.keys(obj).forEach((key: string) => {
    labelMap[key] = Array.isArray(obj[key]) ? obj[key].map((value: MultiSelectOption) => value.value) : obj[key]
  })
  return labelMap
}
