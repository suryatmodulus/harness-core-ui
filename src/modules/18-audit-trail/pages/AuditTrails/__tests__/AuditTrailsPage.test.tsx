import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import { accountPathProps } from '@common/utils/routeUtils'
import * as auditServices from 'services/audit'
import AuditTrailsPage from '../AuditTrailsPage'
import { filters } from '../../../components/__tests__/mockData'

jest.spyOn(auditServices, 'useGetFilterList').mockImplementation(() => ({ data: filters, loading: false } as any))

describe('Audit trail Page', () => {
  test('render', () => {
    const renderObj = render(
      <TestWrapper path={routes.toAuditTrail({ ...accountPathProps })} pathParams={{ accountId: 'testAcc' }}>
        <AuditTrailsPage />
      </TestWrapper>
    )
    expect(renderObj.container).toMatchSnapshot()
  })
})
