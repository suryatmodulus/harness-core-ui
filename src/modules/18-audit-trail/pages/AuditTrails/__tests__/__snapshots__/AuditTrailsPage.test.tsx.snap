// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit trail Page render 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="PageHeader--breadcrumbsDiv"
      >
        <div
          class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <a
              class="Breadcrumbs--breadcrumb"
              href="/account/testAcc/settings"
            >
              <span
                class="bp3-icon bp3-icon-cog StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary7"
                icon="cog"
              >
                <svg
                  data-icon="cog"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    cog
                  </desc>
                  <path
                    d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
              >
                common.accountSettings
              </p>
            </a>
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
              data-icon="main-chevron-right"
            >
              <svg
                height="8"
                viewBox="0 0 8 14"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
        >
          account common.auditTrail
        </p>
      </div>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main PageSubHeader--container subHeaderContainer StyledProps--flex"
  >
    <div
      class="Layout--horizontal StyledProps--main subHeader StyledProps--flex"
    >
      <span
        class="bp3-popover-wrapper"
      >
        <span
          class="bp3-popover-target"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main dateRange Button--with-current-color Button--without-shadow Button--withRightIcon"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              common.last7days
            </span>
            <span
              class="bp3-icon bp3-icon-calendar StyledProps--main"
              icon="calendar"
            >
              <svg
                data-icon="calendar"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  calendar
                </desc>
                <path
                  d="M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder DropDown--minWidth StyledProps--flex"
                data-testid="filter-select"
                style="min-width: 220px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    filters.selectFilter
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="filterButtonContainer"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main ngFilter StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            id="ngfilterbtn"
            type="button"
          >
            <span
              class="bp3-icon StyledProps--main"
              data-icon="ng-filter"
            >
              <svg
                height="22"
                viewBox="0 0 24 24"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M16.5 8V7H19a.5.5 0 000-1h-2.5V5a.5.5 0 00-1 0v1H5a.5.5 0 000 1h10.5v1a.5.5 0 001 0zM5 11.5h3v-1a.5.5 0 011 0v1h10a.5.5 0 010 1H9v1a.5.5 0 01-1 0v-1H5a.5.5 0 010-1zM5 17a.5.5 0 000 1h7v1a.5.5 0 001 0v-1h6a.5.5 0 000-1h-6v-1a.5.5 0 00-1 0v1H5z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        />
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody pageBody"
    data-name="page-body"
  >
    <div
      class="StyledProps--font StyledProps--main PageSpinner--spinner StyledProps--flex StyledProps--flex-align-center-center"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main PageSpinner--content"
        style="align-items: center; width: 500px;"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
          data-icon="steps-spinner"
        >
          <svg
            height="32"
            viewBox="0 0 128 128"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                fill="#ff8e02"
              />
              <path
                d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                fill="#ffddb3"
                fill-opacity="0.3"
              />
              <path
                d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                fill="#ffd29a"
                fill-opacity="0.4"
              />
              <path
                d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                fill="#ffc680"
                fill-opacity="0.5"
              />
              <path
                d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                fill="#ffbb67"
                fill-opacity="0.6"
              />
              <path
                d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                fill="#ffb04d"
                fill-opacity="0.7"
              />
              <path
                d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                fill="#ffa535"
                fill-opacity="0.8"
              />
              <path
                d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                fill="#ff991b"
                fill-opacity="0.9"
              />
              <animatetransform
                attributeName="transform"
                calcMode="discrete"
                dur="1200ms"
                repeatCount="indefinite"
                type="rotate"
                values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
              />
            </g>
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
        >
          Loading, please wait...
        </p>
      </div>
    </div>
  </div>
</div>
`;
